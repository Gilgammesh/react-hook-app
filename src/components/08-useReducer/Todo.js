import React from "react";
import PropTypes from "prop-types";

const Todo = ({ ele, index, dispatch }) => {
  return (
    <div className="col-sm-6 mt-3">
      <div className="card">
        <div className="card-body">
          <h5 className="card-title">ID - {ele.id}</h5>
          <p className="card-text">
            {index + 1}.- {ele.desc}
          </p>
          <p className="card-text">
            {ele.done ? (
              <span className="badge badge-success">Hecho</span>
            ) : (
              <span className="badge badge-warning">Pendiente</span>
            )}
          </p>
          <button
            className="btn btn-danger"
            onClick={() => dispatch({ type: "delete", payload: ele.id })}
          >
            Eliminar
          </button>
        </div>
      </div>
    </div>
  );
};

Todo.propTypes = {
  ele: PropTypes.object.isRequired,
  index: PropTypes.number.isRequired,
  dispatch: PropTypes.func.isRequired,
};

export default Todo;
