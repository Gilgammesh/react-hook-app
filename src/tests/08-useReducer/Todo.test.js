import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import Todo from "../../components/08-useReducer/Todo";

describe("Pruebas unitarias en el componente Todo.js", () => {
  const ele = {
    id: 1,
    desc: "Aprender React",
    done: true,
  };
  const index = 0;
  const dispatch = jest.fn();
  /************************************************************************************/
  test("se debe mostrar el componente <Todo /> correctamente", () => {
    const wrapper = shallow(
      <Todo ele={ele} index={index} dispatch={dispatch} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  /************************************************************************************/
  test("se debe llamar la función dispatch en el componente", () => {
    const wrapper = shallow(
      <Todo ele={ele} index={index} dispatch={dispatch} />
    );
    const btn = wrapper.find("button");
    btn.simulate("click");
    expect(dispatch).toHaveBeenCalled();
    expect(dispatch).toHaveBeenCalledTimes(1);
    expect(dispatch).toHaveBeenCalledWith({ type: "delete", payload: ele.id });
  });
  /************************************************************************************/
  test("se debe mostrar la información del TODO correctamente en el componente", () => {
    const wrapper = shallow(
      <Todo ele={ele} index={index} dispatch={dispatch} />
    );
    const btn = wrapper.find("button");
    btn.simulate("click");
    const h5 = wrapper.find("h5");
    expect(h5.text().trim()).toBe(`ID - ${ele.id}`);
    const p = wrapper.find("p");
    expect(p.at(0).text().trim()).toBe(`${index + 1}.- ${ele.desc}`);
    const span = wrapper.find("span.badge");
    if (ele.done) {
      expect(span.text().trim()).toBe("Hecho");
    } else {
      expect(span.text().trim()).toBe("Pendiente");
    }
  });
});
