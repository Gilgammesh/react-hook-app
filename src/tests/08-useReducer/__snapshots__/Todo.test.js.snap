// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias en el componente Todo.js se debe llamar la funci√≥n eliminar en el componente 1`] = `
<div
  className="col-sm-6 mt-3"
>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        ID - 
        1
      </h5>
      <p
        className="card-text"
      >
        1
        .- 
        Aprender React
      </p>
      <p
        className="card-text"
      >
        <span
          className="badge badge-success"
        >
          Hecho
        </span>
      </p>
      <button
        className="btn btn-danger"
        onClick={[Function]}
      >
        Eliminar
      </button>
    </div>
  </div>
</div>
`;

exports[`Pruebas unitarias en el componente Todo.js se debe mostrar el componente <Todo /> correctamente 1`] = `
<div
  className="col-sm-6 mt-3"
>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        ID - 
        1
      </h5>
      <p
        className="card-text"
      >
        1
        .- 
        Aprender React
      </p>
      <p
        className="card-text"
      >
        <span
          className="badge badge-success"
        >
          Hecho
        </span>
      </p>
      <button
        className="btn btn-danger"
        onClick={[Function]}
      >
        Eliminar
      </button>
    </div>
  </div>
</div>
`;
